src/include/daos/pool_map.h

```diff
@@ -23,14 +23,18 @@
 
 /**
  * pool component types
+ * target and node are pre-registered and managed by DAOS.
+ * Level 2 to 254 are allocated by control plane based on input from
+ * yaml file.
  */
 typedef enum pool_comp_type {
-	PO_COMP_TP_UNKNOWN	= 0,
-	PO_COMP_TP_ROOT		= 1,
-	PO_COMP_TP_RACK		= 10,
-	PO_COMP_TP_NODE		= 40,
-	PO_COMP_TP_TARGET	= 50,
-	/* TODO: more types */
+	PO_COMP_TP_TARGET	= 0, /** reserved, hard-coded */
+	PO_COMP_TP_RANK		= 1, /** reserved, hard-coded */
+	PO_COMP_TP_MIN		= 2, /** first user-defined domain */
+	PO_COMP_TP_NODE		= 2, /** for test only */
+	PO_COMP_TP_MAX		= 254, /** last user-defined domain */
+	PO_COMP_TP_ROOT		= 255,
+	PO_COMP_TP_END		= 256,
```

Comment:
Hmm, it seems co_type is changed now, so you might need update POOL_MAP_VERSION to POOL_MAP_VER_2 ?

Reply:
Good catch, updated.